packages/core/src/components/tabs/tabs.tsx

```diff
@@ -141,15 +157,72 @@ export class Tabs extends AbstractPureComponent<ITabsProps, ITabsState> {
                     role="tablist"
                 >
                     {tabIndicator}
-                    {tabTitles}
+                    {this.props.overflow ? (
+                        <OverflowList
+                            items={tabTitles}
+                            visibleItemRenderer={this.visibleItemsRenderer}
+                            ref={this.refHandlers.overflowList}
+                            {...this.props.overflowListProps}
+                            overflowRenderer={this.overflowRenderer}
```

Comment:
move this line above the `...` spread so the user can override the default renderer. then you won't have to do crazy logic in there.